Dockerfile.windows

```diff
@@ -1,57 +1,93 @@
-# This file describes the standard way to build Docker, using docker on TP4
+# This file describes the standard way to build Docker, using a docker container on Windows 
+# Server 2016
 #
 # Usage:
 #
-# # Assemble the full dev environment. This is slow the first time.
+# # Assemble the full dev environment. This is slow the first time. Run this from
+# # a directory containing the sources you are validating. For example from 
+# # c:\go\src\github.com\docker\docker
+#
 # docker build -t docker -f Dockerfile.windows .
 #
-# # Mount your source in an interactive container for quick testing:
-# docker run -it -v ${pwd}\bundles:c:/go/src/github.com/docker/docker/bundles docker cmd
 #
+# # Build docker in a container. Run the following from a Windows cmd command prommpt,
+# # replacing c:\built with the directory you want the binaries to be placed on the
+# # host system.
+#
+# docker run --rm -v "c:\built:c:\target" docker sh -c 'cd /c/go/src/github.com/docker/docker; hack/make.sh binary; ec=$?; if [ $ec -eq 0 ]; then robocopy /c/go/src/github.com/docker/docker/bundles/$(cat VERSION)/binary /c/target/binary; fi; exit $ec'
+#
+# Important notes:
+# ---------------
+#
+# Multiple commands in a single powershell RUN command are deliberately not done. This is
+# because PS doesn't have a concept quite like set -e in bash. It would be possible to use 
```

Comment:
@jhowardmsft I tried it with this Dockerfile to test a scenario `RUN powershell -Command fail ; success` and if PowerShell aborts on the first error like `set -e` does on Linux:

```
FROM windowsservercore
RUN powershell -Command $ErrorActionPreference = "Stop"; ls notexist ; ls \windows
CMD [ "cmd.exe" ]
```

But building the Docker image with `docker build -t tst .` I got following error:

```
Step 2 : RUN powershell -Command $ErrorActionPreference = "Stop"; ls notexist ; ls \windows
 ---> Running in 651b858a5211
Stop : The term 'Stop' is not recognized as the name of a cmdlet, function,
script file, or operable program. Check the spelling of the name, or if a path
was included, verify that the path is correct and try again.
At line:1 char:26
+ $ErrorActionPreference = Stop; ls notexist ; ls \windows
+                          ~~~~
    + CategoryInfo          : ObjectNotFound: (Stop:String) [], CommandNotFoun
   dException
    + FullyQualifiedErrorId : CommandNotFoundException
```

IMHO there is a problem with the double quotes inside docker. I think this line is valid. Should I open an issue to investigate double quotes further? /cc @friism
 
But changing it to single quotes work as expected:

```
FROM windowsservercore
RUN powershell -Command $ErrorActionPreference = 'Stop'; ls notexist ; ls \windows
CMD [ "cmd.exe" ]
```

It aborts when the first error occurs:

```
PS C:\Users\vagrant\tst> docker build -t tst .
Sending build context to Docker daemon 2.048 kB
Step 1 : FROM windowsservercore
 ---> 6801d964fda5
Step 2 : RUN powershell -Command $ErrorActionPreference = 'Stop'; ls notexist ; ls \windows
 ---> Running in dbc56c10dc07
ls : Cannot find path 'C:\Windows\system32\notexist' because it does not exist.
At line:1 char:34
+ $ErrorActionPreference = 'Stop'; ls notexist ; ls \windows
+                                  ~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Windows\system32\notexist:St
   ring) [Get-ChildItem], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetCh
   ildItemCommand

The command 'cmd /S /C powershell -Command $ErrorActionPreference = 'Stop'; ls notexist ; ls \windows' returned a non-zero code: 1
```